libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -479,7 +479,7 @@ int WiFiClientSecureCtx::_run_until(unsigned target, bool blocking) {
   esp8266::polledTimeout::oneShotMs loopTimeout(_timeout);
 
   for (int no_work = 0; blocking || no_work < 2;) {
-    optimistic_yield(100);
+    optimistic_yield(1000);
```

Comment:
I am undecided on this one. I don't think a forced plain `yield()` is in order here, but I don't quite dare to shot for the 10000µs interval either. If anyone with better insight into what `_run_until` actually does could comment, please?

Reply:
This loop pumps the BSSL engine.  It really should be a small optimistic_yield because the way things work we may initially be able to go thru everything and get a result in 1 pass, in which case the smallest delay is best.  Or, we might need to wait for data to come in over the WiFi to feed in/out of the BSSL engine, in which case we want to `yield()` as there's nothing we can do.

I would not increase the optimistic to 1ms, as when it's waiting and you don't have a yield then you're really just wasting time and no progress can be made until that `yield()` happens and gets some data.

In fact, a `yield()` here would work as well, if you want to scrub out optimistic yields.

Reply:
I'm reverting this spot to the time-proven 100µs. Thanks for the comprehensive input.